gp-includes/template-links.php

```diff
@@ -364,18 +364,33 @@ function gp_link_glossary_delete( $glossary, $set, $text = '', $attrs = array()
  * @param WP_User $user A WP_User user object.
  */
 function gp_link_user( $user ) {
+	// This content is escaped in the gp_link_user_get() method.
+	// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	print( gp_link_user_get( $user ) );
+}
+
+/**
+ * Creates a HTML link to a user profile page.
+ *
+ * @since 4.0.0
+ *
+ * @param WP_User $user A WP_User user object.
+ */
+function gp_link_user_get( $user ): string {
+	$link = '';
 	if ( $user->display_name && $user->display_name !== $user->user_login ) {
-		printf(
+		$link = sprintf(
 			'<a href="%s" tabindex="-1">%s (%s)</a>',
 			esc_url( gp_url_profile( $user->user_nicename ) ),
 			esc_html( $user->display_name ),
 			esc_html( $user->user_login )
 		);
```

Comment:
Could we simplify the logic here?

```suggestion
		return sprintf(
			'<a href="%s" tabindex="-1">%s (%s)</a>',
			esc_url( gp_url_profile( $user->user_nicename ) ),
			esc_html( $user->display_name ),
			esc_html( $user->user_login )
		);
```